src/App.tsx

```diff
@@ -1,61 +1,114 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todos } from './types/todos';
 
 export const App = () => {
+  const [titleError, setTitleError] = useState(false);
+  const [todos, setTodos] = useState(todosFromServer);
+  const [nameError, setnameError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [nameId, setNameId] = useState(0);
+
+  const getId = todos.reduce((acc, todo) => {
+    return Math.max(acc, todo.id);
+  }, -Infinity);
+
+  const reset = () => {
+    setTitle('');
+    setTitleError(false);
+    setnameError(false);
+    setNameId(0);
+  };
+
+  const addTodo = (newTodo: Todos) => {
+    setTodos([...todos, newTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (nameId === 0) {
+      setnameError(true);
+    }
+
+    const onSubmit = {
+      id: getId + 1,
+      title,
+      completed: false,
+      userId: nameId,
+    };
+
+    if (title && nameId !== 0) {
+      addTodo(onSubmit);
+      reset();
+    }
+  };
+
+  const handleName = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setNameId(+event.target.value);
+    setnameError(false);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          Title:&nbsp;
+          <input
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={title}
+            onChange={(event) => {
+              setTitle(event.target.value);
+              setTitleError(false);
+            }}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here